zingo-testutils/src/lightclient.rs

```diff
@@ -0,0 +1,90 @@
+use zcash_client_backend::{PoolType, ShieldedProtocol};
+use zingolib::{error::ZingoLibError, lightclient::LightClient};
+
+/// Create a lightclient from the buffer of another
+pub async fn new_client_from_save_buffer(
+    template_client: &LightClient,
+) -> Result<LightClient, ZingoLibError> {
+    let buffer = template_client.save_internal_buffer().await?;
+
+    LightClient::read_wallet_from_buffer_async(template_client.config(), buffer.as_slice())
+        .await
+        .map_err(ZingoLibError::CantReadWallet)
+}
+/// gets the first address that will allow a sender to send to a specific pool, as a string
+pub async fn get_base_address(client: &LightClient, pooltype: PoolType) -> String {
+    match pooltype {
+        PoolType::Transparent => client.do_addresses().await[0]["receivers"]["transparent"]
+            .clone()
+            .to_string(),
+        PoolType::Shielded(ShieldedProtocol::Sapling) => client.do_addresses().await[0]
+            ["receivers"]["sapling"]
+            .clone()
+            .to_string(),
+        PoolType::Shielded(ShieldedProtocol::Orchard) => {
+            client.do_addresses().await[0]["address"].take().to_string()
+        }
+    }
+}
+/// Helpers to provide raw_receivers to lightclients for send and shield, etc.
+pub mod from_inputs {
+    use zingolib::lightclient::LightClient;
+
+    /// Panics if the address, amount or memo conversion fails.
```

Comment:
doc comment seems incomplete to me, I thought i actually wrote a full comment for this that seems to have got lost

Reply:
I moved some commentary to the top of the mod, and removed the repetition of `test` and `panic` since panic is explicitly still in the remaining doc-comment, and the code now lives in a crate called `testutils`.